examples/flocker/README.md

```diff
@@ -0,0 +1,128 @@
+- *Flocker Container Agent(s)*: a convergence agent that modifies the cluster state to match the desired configuration (unused in this configuration but still required in the cluster).
+
+Read more about the [Flocker Cluster Architecture](https://docs.clusterhq.com/en/latest/concepts/architecture.html) at the [Flocker Documentation](https://docs.clusterhq.com/).
+
+It is recommended to follow [Installing Flocker](https://docs.clusterhq.com/en/latest/install/index.html) and the instructions below to set-up the Flocker cluster to be used with Kubernetes.
+
+#### Flocker Control Service
+
+The Flocker Control Service should be installed manually on a host outside (but accessible from) the Kubernetes cluster. In the future, this may be deployed in pod(s) and exposed as a Kubernetes service.
+
+#### Flocker Agent(s)
+
+The Flocker Agents should be manually installed on *all* Kubernetes nodes. These agents are responsible for provisioning, (de)attachment and (un)mounting and are therefore services that should be run with appropriate privileges on these hosts.
+
+In order for the plugin to connect to Flocker (via REST API), several environment variables must be specified on *all* Kubernetes nodes. This may be specified in an init script for the node's Kubelet service, for example.
+
+- `FLOCKER_CONTROL_SERVICE_HOST` should refer to the hostname of the Control Service
+- `FLOCKER_CONTROL_SERVICE_PORT` should refer to the port of the Control Service (the API service defaults to 4523 but this must still be specified)
+
+The following environment variables should refer to keys and certificates on the host that are specific to that host.
+
+- `FLOCKER_CONTROL_SERVICE_CA_FILE` should refer to the full path to the cluster certificate file
+- `FLOCKER_CONTROL_SERVICE_CLIENT_KEY_FILE` should refer to the full path to the key file for the API user
+- `FLOCKER_CONTROL_SERVICE_CLIENT_CERT_FILE` should refer to the full path to the certificate file for the API user
+
+More details regarding cluster authentication can be found at the documentation: [Flocker Cluster Security & Authentication](https://docs.clusterhq.com/en/latest/concepts/security.html) and [Configuring Cluster Authentication](https://docs.clusterhq.com/en/latest/config/configuring-authentication.html).
+
+### Create a pod with a Flocker volume
+
+The following *volume* spec in  illustrates a sample configuration using a nginx pod with a www-root that uses the volume.
+
+```yaml
+  volumes:
+    - name: www-root
+      flocker:
+        datasetName: my-flocker-vol
+```
+
+The parameters are explained as the followings.
+
+- **datasetName** is the unique name for the Flocker dataset.
+
+If the dataset does not already exist in Flocker, it is automatically provisioned, attached and mounted by the plugin. Otherwise, with an existing dataset, it is reattached by Flocker to the node that the pod is scheduled.
```

Comment:
Yeah, auto-provisioning probably should not happen here.  With acknowledgement that flocker is a somewhat different model than we've seen in the past, I think the simpler thing to do right now is to error out if the volume does not exist, and then we can tie into the nascent provisioning infrastructure.

What think?